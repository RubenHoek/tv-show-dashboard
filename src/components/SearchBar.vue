<script setup>
import { ref } from "vue";

defineProps({
  placeholder: {
    type: String,
    default: "Search",
  },
});

const input = ref("");
</script>

<template>
  <p class="control has-icons-left">
    <input
      class="input"
      type="text"
      :placeholder="placeholder"
      v-model="input"
      @input="$emit('update:modelValue', $event.target.value)"
      @keyup.enter="$emit('enter')"
    />
    <span class="icon is-left">
      <i class="fas fa-search" aria-hidden="true"></i>
    </span>
  </p>
</template>
