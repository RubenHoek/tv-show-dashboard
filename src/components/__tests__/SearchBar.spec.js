import { it, describe, expect } from "vitest";
import { mount } from "@vue/test-utils";
import SearchBar from "../SearchBar.vue";

import { createTestingPinia } from "@pinia/testing";

//Mock props
const placeholder = "Test placholder";

describe.concurrent("Render the SearchBar", () => {
  it("renders the SearchBar", () => {
    const wrapper = mount(SearchBar, {
      props: { placeholder },
      global: {
        plugins: [createTestingPinia({ createSpy: () => {} })],
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
