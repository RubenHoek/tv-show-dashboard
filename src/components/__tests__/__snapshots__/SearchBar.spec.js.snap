// Vitest Snapshot v1

exports[`Render the SearchBar > renders the SearchBar 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <p
        class="control has-icons-left"
      >
        <input
          class="input"
          placeholder="Test placholder"
          type="text"
        />
        <span
          class="icon is-left"
        >
          <i
            aria-hidden="true"
            class="fas fa-search"
          />
        </span>
      </p>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {
          "$pinia": {
            "_a": [Circular],
            "_e": EffectScope {
              "active": true,
              "cleanups": [],
              "detached": true,
              "effects": [],
              "parent": undefined,
            },
            "_p": [
              [Function],
              [Function],
              [Function],
            ],
            "_s": Map {},
            "_testing": true,
            "app": [Circular],
            "install": [Function],
            "state": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": {},
              "_value": {},
              "dep": undefined,
            },
            "use": [Function],
          },
        },
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {
        Symbol(pinia): {
          "_a": [Circular],
          "_e": EffectScope {
            "active": true,
            "cleanups": [],
            "detached": true,
            "effects": [],
            "parent": undefined,
          },
          "_p": [
            [Function],
            [Function],
            [Function],
          ],
          "_s": Map {},
          "_testing": true,
          "app": [Circular],
          "install": [Function],
          "state": RefImpl {
            "__v_isRef": true,
            "__v_isShallow": false,
            "_rawValue": {},
            "_value": {},
            "dep": undefined,
          },
          "use": [Function],
        },
      },
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {
            "$pinia": {
              "_a": [Circular],
              "_e": EffectScope {
                "active": true,
                "cleanups": [],
                "detached": true,
                "effects": [],
                "parent": undefined,
              },
              "_p": [
                [Function],
                [Function],
                [Function],
              ],
              "_s": Map {},
              "_testing": true,
              "app": [Circular],
              "install": [Function],
              "state": RefImpl {
                "__v_isRef": true,
                "__v_isShallow": false,
                "_rawValue": {},
                "_value": {},
                "dep": undefined,
              },
              "use": [Function],
            },
          },
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {
          Symbol(pinia): {
            "_a": [Circular],
            "_e": EffectScope {
              "active": true,
              "cleanups": [],
              "detached": true,
              "effects": [],
              "parent": undefined,
            },
            "_p": [
              [Function],
              [Function],
              [Function],
            ],
            "_s": Map {},
            "_testing": true,
            "app": [Circular],
            "install": [Function],
            "state": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": {},
              "_value": {},
              "dep": undefined,
            },
            "use": [Function],
          },
        },
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {
                "$pinia": {
                  "_a": [Circular],
                  "_e": EffectScope {
                    "active": true,
                    "cleanups": [],
                    "detached": true,
                    "effects": [],
                    "parent": undefined,
                  },
                  "_p": [
                    [Function],
                    [Function],
                    [Function],
                  ],
                  "_s": Map {},
                  "_testing": true,
                  "app": [Circular],
                  "install": [Function],
                  "state": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": {},
                    "_value": {},
                    "dep": undefined,
                  },
                  "use": [Function],
                },
              },
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {
              Symbol(pinia): {
                "_a": [Circular],
                "_e": EffectScope {
                  "active": true,
                  "cleanups": [],
                  "detached": true,
                  "effects": [],
                  "parent": undefined,
                },
                "_p": [
                  [Function],
                  [Function],
                  [Function],
                ],
                "_s": Map {},
                "_testing": true,
                "app": [Circular],
                "install": [Function],
                "state": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": {},
                  "_value": {},
                  "dep": undefined,
                },
                "use": [Function],
              },
            },
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "placeholder": "Test placholder",
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "input": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": "",
              "_value": "",
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Set {
                      [Circular],
                    },
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
            },
            "ref": [Function],
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {
            "hasOwnProperty": [Function],
          },
          "exposed": {
            "hasOwnProperty": [Function],
          },
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "placeholder": "Test placholder",
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "placeholder": {
                "0": false,
                "1": true,
                "default": "Search",
                "type": [Function],
              },
            },
            [
              "placeholder",
            ],
          ],
          "provides": {},
          "proxy": {
            "placeholder": "Test placholder",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [
            [Function],
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "input": "",
            "ref": [Function],
          },
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": [
                  {
                    "arg": undefined,
                    "dir": {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "mounted": [Function],
                    },
                    "instance": {
                      "hasOwnProperty": [Function],
                    },
                    "modifiers": {},
                    "oldValue": undefined,
                    "value": "",
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": [
                  "placeholder",
                ],
                "el": <input
                  class="input"
                  placeholder="Test placholder"
                  type="text"
                />,
                "key": null,
                "patchFlag": 40,
                "props": {
                  "class": "input",
                  "onInput": [Function],
                  "onKeyup": [Function],
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Test placholder",
                  "type": "text",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <i
                      aria-hidden="true"
                      class="fas fa-search"
                    />,
                    "key": null,
                    "patchFlag": 0,
                    "props": {
                      "aria-hidden": "true",
                      "class": "fas fa-search",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "i",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <span
                  class="icon is-left"
                >
                  <i
                    aria-hidden="true"
                    class="fas fa-search"
                  />
                </span>,
                "key": null,
                "patchFlag": -1,
                "props": {
                  "class": "icon is-left",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "span",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": [
                  {
                    "arg": undefined,
                    "dir": {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "mounted": [Function],
                    },
                    "instance": {
                      "hasOwnProperty": [Function],
                    },
                    "modifiers": {},
                    "oldValue": undefined,
                    "value": "",
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": [
                  "placeholder",
                ],
                "el": <input
                  class="input"
                  placeholder="Test placholder"
                  type="text"
                />,
                "key": null,
                "patchFlag": 40,
                "props": {
                  "class": "input",
                  "onInput": [Function],
                  "onKeyup": [Function],
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Test placholder",
                  "type": "text",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
            ],
            "dynamicProps": null,
            "el": <p
              class="control has-icons-left"
            >
              <input
                class="input"
                placeholder="Test placholder"
                type="text"
              />
              <span
                class="icon is-left"
              >
                <i
                  aria-hidden="true"
                  class="fas fa-search"
                />
              </span>
            </p>,
            "key": null,
            "patchFlag": 0,
            "props": {
              "class": "control has-icons-left",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "p",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "C:/Projecten/Mijn projecten/ABN Assignment/tv-show-dashboard/src/components/SearchBar.vue",
            "__name": "SearchBar",
            "components": {},
            "props": {
              "placeholder": {
                "0": false,
                "1": true,
                "default": "Search",
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <p
          class="control has-icons-left"
        >
          <input
            class="input"
            placeholder="Test placholder"
            type="text"
          />
          <span
            class="icon is-left"
          >
            <i
              aria-hidden="true"
              class="fas fa-search"
            />
          </span>
        </p>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "placeholder": "Test placholder",
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "C:/Projecten/Mijn projecten/ABN Assignment/tv-show-dashboard/src/components/SearchBar.vue",
          "__name": "SearchBar",
          "components": {},
          "props": {
            "placeholder": {
              "0": false,
              "1": true,
              "default": "Search",
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {
              "$pinia": {
                "_a": [Circular],
                "_e": EffectScope {
                  "active": true,
                  "cleanups": [],
                  "detached": true,
                  "effects": [],
                  "parent": undefined,
                },
                "_p": [
                  [Function],
                  [Function],
                  [Function],
                ],
                "_s": Map {},
                "_testing": true,
                "app": [Circular],
                "install": [Function],
                "state": RefImpl {
                  "__v_isRef": true,
                  "__v_isShallow": false,
                  "_rawValue": {},
                  "_value": {},
                  "dep": undefined,
                },
                "use": [Function],
              },
            },
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {
            Symbol(pinia): {
              "_a": [Circular],
              "_e": EffectScope {
                "active": true,
                "cleanups": [],
                "detached": true,
                "effects": [],
                "parent": undefined,
              },
              "_p": [
                [Function],
                [Function],
                [Function],
              ],
              "_s": Map {},
              "_testing": true,
              "app": [Circular],
              "install": [Function],
              "state": RefImpl {
                "__v_isRef": true,
                "__v_isShallow": false,
                "_rawValue": {},
                "_value": {},
                "dep": undefined,
              },
              "use": [Function],
            },
          },
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <p
          class="control has-icons-left"
        >
          <input
            class="input"
            placeholder="Test placholder"
            type="text"
          />
          <span
            class="icon is-left"
          >
            <i
              aria-hidden="true"
              class="fas fa-search"
            />
          </span>
        </p>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {
        "$pinia": {
          "_a": [Circular],
          "_e": EffectScope {
            "active": true,
            "cleanups": [],
            "detached": true,
            "effects": [],
            "parent": undefined,
          },
          "_p": [
            [Function],
            [Function],
            [Function],
          ],
          "_s": Map {},
          "_testing": true,
          "app": [Circular],
          "install": [Function],
          "state": RefImpl {
            "__v_isRef": true,
            "__v_isShallow": false,
            "_rawValue": {},
            "_value": {},
            "dep": undefined,
          },
          "use": [Function],
        },
      },
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.41",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {
    "placeholder": "Test placholder",
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <p
    class="control has-icons-left"
  >
    <input
      class="input"
      placeholder="Test placholder"
      type="text"
    />
    <span
      class="icon is-left"
    >
      <i
        aria-hidden="true"
        class="fas fa-search"
      />
    </span>
  </p>,
}
`;
